### PR TITLE
chore(masthead): create Redux store

### PR DESCRIPTION
**Pull request by asudoh**
### Description

This change creates a Redux store that works with `LocaleAPI` and `TranslateAPI`, and connect it to `<dds-masthead-container>`.

Such change has the following merits:

1. The procedure of loading locale data and translation data, with caching mechanism and error handling mechanism in place, can be reused for multiple components and all framework variants.
   (The plan is moving the Redux code to `@carbon/ibmdotcom-services` or a new package once the Redux store is stabilized enough)
2. Lighter-weight component code (of `<dds-masthead-container>`).
   Notably much less lifecycle management, given we no longer have to deal with stale async requests for unmounted `<dds-masthead-container>` instance.

### Changelog

**New**

- A Redux store to work with `LocaleAPI` and `TranslateAPI`.

**Removed**

- Code in `<dds-masthead-container>` code that works with `LocaleAPI` and `TranslateAPI`.

### CODE REVIEW

**Comment by jeffchew** [comment]
Hey @asudoh ! 

I'm looking through and have some hesitation on adding redux to this. Seems to be a lot of additional overhead to maintain if for example any of the services changed. We are already doing some level of caching on the services side, especially for locale/translation data that is stored in localStorage and the service facade itself can manage how that data gets passed out. 

Throwing this out there for anyone else that has thoughts on this, @kennylam @annawen1 @RaphaelAmadeu @andysherman2121

**Comment by asudoh** [comment]
Thank you for your feedback @jeffchew, eliminated most of the overhead by abstracting the repetitive code in actions, thunk actions and reducers (lots of red portions in [`actions/localeAPI.ts`](https://github.com/carbon-design-system/ibm-dotcom-library/pull/3197/commits/2146a4b33956fd7a8d980a13407805a4c9eddb2e#diff-1cf9d5861af32727c626486221b8bfbc), [`actions/translateAPI.ts`](https://github.com/carbon-design-system/ibm-dotcom-library/pull/3197/commits/2146a4b33956fd7a8d980a13407805a4c9eddb2e#diff-d6c1fb4be567c21eb44508fe9ebe6d79), [`reducers/localeAPI.ts`](https://github.com/carbon-design-system/ibm-dotcom-library/pull/3197/commits/2146a4b33956fd7a8d980a13407805a4c9eddb2e#diff-ab591c6ca7f260d80d723e178ecb8dcd) and [`reducers/translateAPI.ts`](https://github.com/carbon-design-system/ibm-dotcom-library/pull/3197/commits/2146a4b33956fd7a8d980a13407805a4c9eddb2e#diff-a55fe7549da49086a59683fb91f8a987)). Wondering this change makes sense to you.

**Comment by jeffchew** [comment]
> Thank you for your feedback @jeffchew, eliminated most of the overhead by abstracting the repetitive code in actions, thunk actions and reducers (lots of red portions in [`actions/localeAPI.ts`](https://github.com/carbon-design-system/ibm-dotcom-library/pull/3197/commits/2146a4b33956fd7a8d980a13407805a4c9eddb2e#diff-1cf9d5861af32727c626486221b8bfbc), [`actions/translateAPI.ts`](https://github.com/carbon-design-system/ibm-dotcom-library/pull/3197/commits/2146a4b33956fd7a8d980a13407805a4c9eddb2e#diff-d6c1fb4be567c21eb44508fe9ebe6d79), [`reducers/localeAPI.ts`](https://github.com/carbon-design-system/ibm-dotcom-library/pull/3197/commits/2146a4b33956fd7a8d980a13407805a4c9eddb2e#diff-ab591c6ca7f260d80d723e178ecb8dcd) and [`reducers/translateAPI.ts`](https://github.com/carbon-design-system/ibm-dotcom-library/pull/3197/commits/2146a4b33956fd7a8d980a13407805a4c9eddb2e#diff-a55fe7549da49086a59683fb91f8a987)). Wondering this change makes sense to you.

Thank you for the updates @asudoh ! I actually added redux and web components to the huddle topic for tomorrow, so we can discuss as a group. Want to make sure we are all aligned and if there are any other considerations we need to think about for this.

File: packages/web-components/src/globals/mixins/connect.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ */
+
+import { Dispatch, Store } from 'redux';
+import Handle from '../internal/handle';
+
+/**
+ * @param store A redux store.
+ * @returns A funciton that takes a base class and returns a mix-in that connects the component to Redux store.
+ */
+const ConnectMixin = <TState, TStateProps = { [name: string]: any }, TDispatchProps = { [name: string]: any }>(
```
Comment (jeffchew) [inline packages/web-components/src/globals/mixins/connect.ts:17]: Not sure if there's an overlap, wondering if we wanted to use some of the redux mixins from polymer? https://github.com/Polymer/pwa-helpers#redux-helpers

Comment (asudoh) [reply packages/web-components/src/globals/mixins/connect.ts:17]: @jeffchew I took a look at it before I implemented it myself - And found our that there was no `mapStateToProps()` support (and thus a custom implementation).

**Comment by asudoh** [comment]
Thank you for the opportunity to discuss about Redux @jeffchew! Put this PR back to the original state as discussed, and let's plan on moving the caching stuffs to services API and creating a separate package for the Redux logic (or putting it to services package).

**Comment by jeffchew** [comment]
@asudoh was checking the percy snapshots for the masthead, is this using the service call or manual data for the navigation? https://percy.io/538fc19a/IBM.com-Library-Web-Components/builds/6204180

For react, we have it using the manual data since waiting for the service call can be flakey when uploading the snapshots to percy, one of those where we use the `inPercy()` logic:

https://github.com/carbon-design-system/ibm-dotcom-library/blob/75735770efef629f2c5695d5fbec6a644579725b/packages/react/src/components/Masthead/__stories__/Masthead.stories.js#L51

**Comment by asudoh** [comment]
@jeffchew It's hybrid, if `CORS_PROXY` is set up it uses a live request, otherwise it uses mock data. And tracking it in #3242 sounds good - Thanks!

**Comment by asudoh** [comment]
Created: #3258

**Pull request merged**